byob.js

```diff
@@ -292,11 +292,17 @@ CustomBlockDefinition.prototype.inputOptionsOfIdx = function (idx) {
 };
 
 CustomBlockDefinition.prototype.dropDownMenuOf = function (inputName) {
-    var dict = {};
+    var dict = {},
+	    gensym = 1;
```

Comment:
This should be a real word. No need to save characters. :-) In this case I'm not sure what it means.

(This is one habit that's been drilled into me the past few moths, a bit painfully, but I think it definitely ends up helping out.)

Otherwise, this little algorithm could use a comment a why/when we need to check of uniqueness.